be/src/exprs/agg/percentile_cont.h

```diff
@@ -0,0 +1,172 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include <cstring>
+#include <limits>
+#include <type_traits>
+
+#include "column/column_helper.h"
+#include "column/object_column.h"
+#include "column/vectorized_fwd.h"
+#include "exprs/agg/aggregate.h"
+#include "gutil/casts.h"
+
+namespace starrocks::vectorized {
+
+// AvgResultPT for final result
+template <PrimitiveType PT, typename = guard::Guard>
+inline constexpr PrimitiveType PercentileResultPT = PT;
+
+template <PrimitiveType PT>
+inline constexpr PrimitiveType PercentileResultPT<PT, ArithmeticPTGuard<PT>> = TYPE_DOUBLE;
+
+template <PrimitiveType PT, typename = guard::Guard>
+struct PercentileContState {
+    using CppType = RunTimeCppType<PT>;
+    void update(CppType item) { items.emplace_back(item); }
+    void sort() { std::stable_sort(std::begin(items), std::end(items)); }
+
+    std::vector<CppType> items;
+    double rate = 0.0;
+};
+
+template <PrimitiveType PT>
+class PercentileContAggregateFunction final
```

Comment:
To make our system compile fast, we'd better to make these code in source files other than header files.

Reply:
We have to use the header file, because this kind of function is a template class and cannot be placed in the cpp file